service/src/java/org/apache/hive/service/cli/HiveSQLException.java

```diff
@@ -102,149 +115,39 @@ public HiveSQLException(String reason, String sqlState, int vendorCode, Throwabl
   }
 
   public HiveSQLException(TStatus status) {
-    // TODO: set correct vendorCode field
     super(status.getErrorMessage(), status.getSqlState(), status.getErrorCode());
-    if (status.getInfoMessages() != null) {
-      initCause(toCause(status.getInfoMessages()));
-    }
   }
 
-
-
-/**
-   * Converts current object to a {@link TStatus} object
-   * @return	a {@link TStatus} object
+  /**
+   * Converts current object to a {@link TStatus} object.
+   *
+   * @return a {@link TStatus} object
    */
   public TStatus toTStatus() {
     // TODO: convert sqlState, etc.
     TStatus tStatus = new TStatus(TStatusCode.ERROR_STATUS);
     tStatus.setSqlState(getSQLState());
     tStatus.setErrorCode(getErrorCode());
     tStatus.setErrorMessage(getMessage());
-    tStatus.setInfoMessages(toString(this));
+    tStatus.setInfoMessages(DEFAULT_INFO);
     return tStatus;
   }
 
   /**
-   * Converts the specified {@link Exception} object into a {@link TStatus} object
-   * @param e	a {@link Exception} object
-   * @return	a {@link TStatus} object
+   * Converts the specified {@link Exception} object into a {@link TStatus}
+   * object.
+   *
+   * @param e a {@link Exception} object
+   * @return a {@link TStatus} object
    */
   public static TStatus toTStatus(Exception e) {
     if (e instanceof HiveSQLException) {
-      return ((HiveSQLException)e).toTStatus();
+      return ((HiveSQLException) e).toTStatus();
     }
     TStatus tStatus = new TStatus(TStatusCode.ERROR_STATUS);
     tStatus.setErrorMessage(e.getMessage());
-    tStatus.setInfoMessages(toString(e));
+    tStatus.setInfoMessages(DEFAULT_INFO);
```

Comment:
Basically we will throw a general message instead of explicit error message? Is that because we already have such information in ErrorMessage?

Reply:
So, the error message itself is captured in `setErrorMessage`.  The `Info Messages` is currently populated with the Exception stacktrace.  We should not be sending such internal details to the client.  Removing it altogether would be too big of a change, so I just put a static message in there for now as filler.

Reply:
Make sense.